lsp/src/service/registry/registry.ts

```diff
@@ -26,6 +26,13 @@ export class LanguageServiceRegistry implements LanguageServiceContext {
     }
 
     private async getMatchingStrategy(textDocument: TextDocument): Promise<LanguageServiceStrategy | undefined> {
-        return Promise.any(this.strategies.filter(strategy => strategy.isMatch(textDocument)))
+        for (const strategy of this.strategies) {
+            const isMatch = await strategy.isMatch(textDocument)
```

Comment:
What is the difference vs `strategies.filter` ? Maybe just `Promise.any` isn't needed?

Reply:
> The updated code also exits after the first successful match. We don't need to run all of the remaining evaluations.

sure, can use `some()` instead of filter()